vortex-scalar/src/struct_.rs

```diff
@@ -80,17 +80,14 @@ impl<'a> StructScalar<'a> {
     }
 
     pub fn field_by_idx(&self, idx: usize) -> VortexResult<Scalar> {
-        let DType::Struct(st, nullability) = self.dtype() else {
+        let DType::Struct(st, _) = self.dtype() else {
             unreachable!()
         };
 
-        let mut field_dtype = st.field_by_index(idx)?;
-        if matches!(nullability, Nullability::Nullable) {
-            field_dtype = field_dtype.with_nullability(Nullability::Nullable);
-        }
+        let field_dtype = st.field_by_index(idx)?;
 
         Ok(match self.fields {
-            None => Scalar::null(field_dtype),
+            None => Scalar::null(field_dtype.as_nullable()),
```

Comment:
I don't think this is right.

If the StructScalar itself is null, then you shouldn't be able to get a field from it at all right? This should panic and the user should have checked is_null() beforehand?

Or we can return Option<Scalar>, but it's a bit more confusing IMO.

Reply:
I think panicing here is maybe more obvious.

Reply:
It's the same behaviour as unwrapping I guess.

Can always add the field_opt function if we need one later